plugins/Goals/Archiver.php

```diff
@@ -450,28 +474,30 @@ public function aggregateMultipleReports()
         /*
          * Archive Ecommerce Items
          */
-        $dataTableToSum = $this->dimensionRecord;
-        foreach ($this->dimensionRecord as $recordName) {
-            $dataTableToSum[] = self::getItemRecordNameAbandonedCart($recordName);
-        }
-        $columnsAggregationOperation = null;
+        if (Manager::getInstance()->isPluginActivated('Ecommerce')) {
+            $dataTableToSum = $this->dimensionRecord;
+            foreach ($this->dimensionRecord as $recordName) {
+                $dataTableToSum[] = self::getItemRecordNameAbandonedCart($recordName);
+            }
+            $columnsAggregationOperation = null;
 
-        $this->getProcessor()->aggregateDataTableRecords($dataTableToSum,
-            $maximumRowsInDataTableLevelZero = $this->productReportsMaximumRows,
-            $maximumRowsInSubDataTable = $this->productReportsMaximumRows,
-            $columnToSortByBeforeTruncation = Metrics::INDEX_ECOMMERCE_ITEM_REVENUE,
-            $columnsAggregationOperation,
-            $columnsToRenameAfterAggregation = null,
-            $countRowsRecursive = array());
+            $this->getProcessor()->aggregateDataTableRecords($dataTableToSum,
+                $maximumRowsInDataTableLevelZero = $this->productReportsMaximumRows,
+                $maximumRowsInSubDataTable = $this->productReportsMaximumRows,
+                $columnToSortByBeforeTruncation = Metrics::INDEX_ECOMMERCE_ITEM_REVENUE,
+                $columnsAggregationOperation,
+                $columnsToRenameAfterAggregation = null,
+                $countRowsRecursive = []);
+        }
 
         /*
          *  Archive General Goal metrics
          */
         $goalIdsToSum = GoalManager::getGoalIds($this->getProcessor()->getParams()->getSite()->getId());
 
         //Ecommerce
-        $goalIdsToSum[] = GoalManager::IDGOAL_ORDER;
-        $goalIdsToSum[] = GoalManager::IDGOAL_CART; //bug here if idgoal=1
+        $goalIdsToSum = array_merge($goalIdsToSum, $this->getEcommerceIdGoals());
```

Comment:
should we only add the ecommerceGoalIds if ecommerce is enabled? and then `$this->getProcessor()->aggregateNumericMetrics($fieldsToSum);` we could run only if `$fieldsToSum` is not empty? $this->getProcessor()->aggregateNumericMetrics($fieldsToSum);

Reply:
I have added a check for ecommerce plugin. But the `$goalIdsToSum` array actually always contains the entry `false`, for processing the overview. Not sure if we should skip that one if no other goal is processed or not.

Reply:
Guess we could always keep the `false` 👍